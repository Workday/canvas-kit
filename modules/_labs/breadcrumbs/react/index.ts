import Breadcrumbs from './lib/Breadcrumbs';

export default Breadcrumbs;
export {Breadcrumbs};
export * from './lib/Breadcrumbs';
