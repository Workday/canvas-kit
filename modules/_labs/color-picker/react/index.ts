export {default, default as ColorPicker, ColorPickerProps} from './lib/ColorPicker';
export {Swatch} from './lib/parts/Swatch';
