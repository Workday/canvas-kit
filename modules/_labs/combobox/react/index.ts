import Combobox from './lib/Combobox';

export default Combobox;
export {Combobox};
export * from './lib/Combobox';
