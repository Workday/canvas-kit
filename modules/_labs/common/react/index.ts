export * from './lib/Flex';
export * from './lib/Box';
