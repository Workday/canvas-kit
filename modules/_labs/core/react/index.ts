import type from './lib/type';
import space from './lib/space';
export default type;
export {type, space};
export * from './lib/type';
export * from './lib/StaticStates';
