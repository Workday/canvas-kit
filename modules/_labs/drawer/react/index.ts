import Drawer from './lib/Drawer';
import DrawerHeader from './lib/DrawerHeader';
export {Drawer, DrawerHeader};
export default Drawer;
export * from './lib/Drawer';
