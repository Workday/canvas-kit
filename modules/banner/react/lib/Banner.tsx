import * as React from 'react';
import styled from '@emotion/styled';
import {colors, spacing, borderRadius, type, Themeable} from '@workday/canvas-kit-react-core';
import {SystemIcon} from '@workday/canvas-kit-react-icon';
import {exclamationCircleIcon, exclamationTriangleIcon} from '@workday/canvas-system-icons-web';
import {ErrorType, focusRing} from '@workday/canvas-kit-react-common';

export enum BannerVariant {
  Full,
  Sticky,
}

export interface BannerProps extends Themeable, React.ButtonHTMLAttributes<HTMLButtonElement> {
  /**
   * Action when interacting with the banner
   */
  onClick?: (e: React.SyntheticEvent) => void;
  /**
   * Text inside the banner
   */
  label?: React.ReactNode;
  /**
   * Size and position of the banner
   */
  variant?: BannerVariant;
  /**
   * Color of the banner based on the type of the error
   */
  error?: ErrorType;
  /**
   * Text on the right, call to action
   */
  actionText?: string;
}

const BannerWrapper = styled('button')<BannerProps>(
  {
    ...type.body2,
    backgroundColor: colors.cantaloupe400,
    boxSizing: 'border-box',
    color: colors.blackPepper400,
    fontSize: 14,
    fontWeight: 500,
    padding: `${spacing.xxs} ${spacing.s}`,
    textAlign: 'left',
    border: 0,
    display: 'flex',
    alignItems: 'center',
    transition: 'background-color 120ms',
    '&:hover': {
      cursor: 'pointer',
    },
  },
  ({error, theme}) => ({
    backgroundColor:
      error === ErrorType.Error ? theme.palette.error.main : theme.palette.alert.main,
    color: error === ErrorType.Error ? theme.palette.error.contrast : colors.blackPepper400,
    '&:hover': {
      backgroundColor:
        error === ErrorType.Error ? theme.palette.error.dark : theme.palette.alert.dark,
    },
    '&:focus': {
      outline: 'none',
      ...focusRing(2, 2, undefined, undefined, undefined, theme.palette.common.focusOutline),
    },
  }),
  ({variant}) => ({
    borderRadius:
      variant === BannerVariant.Sticky ? `${borderRadius.m} 0 0 ${borderRadius.m}` : borderRadius.m,
    width: variant === BannerVariant.Sticky ? '222px' : '328px',
  })
);

const BannerIcon = styled(SystemIcon)({
  marginRight: spacing.xs,
});

const BannerLabel = styled('div')({
  flex: 1,
});

const BannerViewAll = styled('span')<BannerProps>(
  {
    textDecoration: 'underline',
  },
  ({variant}) => ({
    display: variant === BannerVariant.Sticky ? 'none' : undefined,
  })
);

export default class Banner extends React.Component<BannerProps> {
  static Variant = BannerVariant;
  static ErrorType = ErrorType;

  public static defaultProps = {
    actionText: 'View All',
    error: ErrorType.Alert,
    variant: BannerVariant.Full,
  };

  public render() {
    const {label, onClick, actionText, variant, error, ...props} = this.props;

    const bannerIcon = error === ErrorType.Error ? exclamationCircleIcon : exclamationTriangleIcon;
    const iconColor = error === ErrorType.Error ? colors.frenchVanilla100 : colors.blackPepper400;
    const iconSize = 24;

    return (
      <BannerWrapper
        aria-label={actionText + ': ' + label}
        role="button"
        variant={variant}
        tabIndex={0}
        onClick={onClick}
        error={error}
        {...props}
      >
        <BannerIcon icon={bannerIcon} color={iconColor} colorHover={iconColor} size={iconSize} />
        <BannerLabel>{label}</BannerLabel>
        <BannerViewAll variant={variant}>{actionText}</BannerViewAll>
      </BannerWrapper>
    );
  }
}
