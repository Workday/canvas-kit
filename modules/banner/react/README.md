# Canvas Kit Banner

Errors and Alerts notify users of missteps that happen within a workflow and describe how the user
can take appropriate action to resolve them.

## Installation

```sh
yarn add @workday/canvas-kit-react
```

or

```sh
yarn add @workday/canvas-kit-react-banner
```

## Usage

```tsx
import Banner from '@workday/canvas-kit-react-banner';

<Banner label="3 errors" />;
```

## Static Properties

#### `ErrorType: ErrorType`

```tsx
<Banner error={Banner.ErrorType.Error} label="3 errors" />
```

---

#### `Variant: BannerVariant`

```tsx
<Banner variant={Banner.Variant.Sticky} label="3 errors" />
```

## Component Props

### Required

#### `label: String`

> Label of the banner

### Optional

#### `onClick: (e: React.SyntheticEvent) => void`

> Function called when the user click on the banner

#### `variant: BannerVariant`

> Set the banner variant as `full` or `sticky`

Default: `BannerVariant.Full`

#### `error: ErrorType`

> Set the banner type as `alert` or `error`

Default: `ErrorType.Alert`

#### `actionText: string`

> Set the action text in the `full` variant

Default: `'View All'`
