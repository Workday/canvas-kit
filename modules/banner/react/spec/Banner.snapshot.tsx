import * as React from 'react';
import Banner from '../lib/Banner';
import * as renderer from 'react-test-renderer';

describe('Banner Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(<Banner />);
    expect(component).toMatchSnapshot();
  });
  test('renders a full banner as expected', () => {
    const component = renderer.create(<Banner variant={Banner.Variant.Full} />);
    expect(component).toMatchSnapshot();
  });
  test('renders a sticky banner as expected', () => {
    const component = renderer.create(<Banner variant={Banner.Variant.Sticky} />);
    expect(component).toMatchSnapshot();
  });
  test('renders an error banner as expected', () => {
    const component = renderer.create(<Banner error={Banner.ErrorType.Error} />);
    expect(component).toMatchSnapshot();
  });
  test('renders an alert banner as expected', () => {
    const component = renderer.create(<Banner error={Banner.ErrorType.Alert} />);
    expect(component).toMatchSnapshot();
  });
  test('renders a banner with a custom action text expected', () => {
    const component = renderer.create(<Banner actionText="Do an action" />);
    expect(component).toMatchSnapshot();
  });
  test('renders a banner with a custom label expected', () => {
    const component = renderer.create(<Banner label="Hello" />);
    expect(component).toMatchSnapshot();
  });
});
