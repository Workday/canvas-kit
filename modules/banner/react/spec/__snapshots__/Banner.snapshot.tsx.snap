// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Snapshots renders a banner with a custom action text expected 1`] = `
.emotion-6 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 500;
  color: #333333;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  background-color: #ffa126;
  box-sizing: border-box;
  padding: 8px 16px;
  text-align: left;
  border: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: background-color 120ms;
  transition: background-color 120ms;
  background-color: #ffa126;
  color: #333333;
  border-radius: 4px;
  width: 328px;
}

.emotion-6:focus {
  outline: none;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-6:hover {
  cursor: pointer;
}

.emotion-6:hover {
  background-color: #f38b00;
}

.emotion-1 {
  display: inline-block;
  margin-right: 12px;
}

.emotion-1 svg {
  display: block;
}

.emotion-1 .wd-icon-fill {
  fill: #333333;
}

.emotion-1:hover .wd-icon-fill {
  fill: #333333;
}

.emotion-1 .wd-icon-accent {
  fill: #333333;
}

.emotion-1:hover .wd-icon-accent {
  fill: #333333;
}

.emotion-1 .wd-icon-background {
  fill: transparent;
}

.emotion-1:hover .wd-icon-background {
  fill: transparent;
}

.emotion-1 svg {
  width: 24px;
  height: 24px;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-4 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<button
  aria-label="Do an action: undefined"
  className="emotion-6 emotion-7"
  role="button"
  tabIndex={0}
>
  <span
    className="emotion-0 emotion-1"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-exclamation-triangle wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M12.004 6.393L17.995 19H6z\\" class=\\"wd-icon-background\\"/><path d=\\"M12.442 3.316l8.468 16.918a.5.5 0 0 1-.448.724l-16.888-.003a.5.5 0 0 1-.447-.723l8.42-16.915a.5.5 0 0 1 .895-.001zm-.438 3.077L6 19h11.995L12.004 6.393z\\" class=\\"wd-icon-fill\\"/><path d=\\"M12.495 10c.279 0 .505.215.505.498v4.004a.494.494 0 0 1-.505.498h-.99a.496.496 0 0 1-.505-.498v-4.004c0-.275.214-.498.505-.498h.99zm0 8h-.99a.497.497 0 0 1-.505-.505v-.99c0-.291.226-.505.505-.505h.99c.291 0 .505.226.505.505v.99c0 .291-.226.505-.505.505z\\" class=\\"wd-icon-accent\\"/></g></svg>",
      }
    }
  />
  <div
    className="emotion-2 emotion-3"
  />
  <span
    className="emotion-4 emotion-5"
  >
    Do an action
  </span>
</button>
`;

exports[`Banner Snapshots renders a banner with a custom label expected 1`] = `
.emotion-6 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 500;
  color: #333333;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  background-color: #ffa126;
  box-sizing: border-box;
  padding: 8px 16px;
  text-align: left;
  border: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: background-color 120ms;
  transition: background-color 120ms;
  background-color: #ffa126;
  color: #333333;
  border-radius: 4px;
  width: 328px;
}

.emotion-6:focus {
  outline: none;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-6:hover {
  cursor: pointer;
}

.emotion-6:hover {
  background-color: #f38b00;
}

.emotion-1 {
  display: inline-block;
  margin-right: 12px;
}

.emotion-1 svg {
  display: block;
}

.emotion-1 .wd-icon-fill {
  fill: #333333;
}

.emotion-1:hover .wd-icon-fill {
  fill: #333333;
}

.emotion-1 .wd-icon-accent {
  fill: #333333;
}

.emotion-1:hover .wd-icon-accent {
  fill: #333333;
}

.emotion-1 .wd-icon-background {
  fill: transparent;
}

.emotion-1:hover .wd-icon-background {
  fill: transparent;
}

.emotion-1 svg {
  width: 24px;
  height: 24px;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-4 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<button
  aria-label="View All: Hello"
  className="emotion-6 emotion-7"
  role="button"
  tabIndex={0}
>
  <span
    className="emotion-0 emotion-1"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-exclamation-triangle wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M12.004 6.393L17.995 19H6z\\" class=\\"wd-icon-background\\"/><path d=\\"M12.442 3.316l8.468 16.918a.5.5 0 0 1-.448.724l-16.888-.003a.5.5 0 0 1-.447-.723l8.42-16.915a.5.5 0 0 1 .895-.001zm-.438 3.077L6 19h11.995L12.004 6.393z\\" class=\\"wd-icon-fill\\"/><path d=\\"M12.495 10c.279 0 .505.215.505.498v4.004a.494.494 0 0 1-.505.498h-.99a.496.496 0 0 1-.505-.498v-4.004c0-.275.214-.498.505-.498h.99zm0 8h-.99a.497.497 0 0 1-.505-.505v-.99c0-.291.226-.505.505-.505h.99c.291 0 .505.226.505.505v.99c0 .291-.226.505-.505.505z\\" class=\\"wd-icon-accent\\"/></g></svg>",
      }
    }
  />
  <div
    className="emotion-2 emotion-3"
  >
    Hello
  </div>
  <span
    className="emotion-4 emotion-5"
  >
    View All
  </span>
</button>
`;

exports[`Banner Snapshots renders a full banner as expected 1`] = `
.emotion-6 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 500;
  color: #333333;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  background-color: #ffa126;
  box-sizing: border-box;
  padding: 8px 16px;
  text-align: left;
  border: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: background-color 120ms;
  transition: background-color 120ms;
  background-color: #ffa126;
  color: #333333;
  border-radius: 4px;
  width: 328px;
}

.emotion-6:focus {
  outline: none;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-6:hover {
  cursor: pointer;
}

.emotion-6:hover {
  background-color: #f38b00;
}

.emotion-1 {
  display: inline-block;
  margin-right: 12px;
}

.emotion-1 svg {
  display: block;
}

.emotion-1 .wd-icon-fill {
  fill: #333333;
}

.emotion-1:hover .wd-icon-fill {
  fill: #333333;
}

.emotion-1 .wd-icon-accent {
  fill: #333333;
}

.emotion-1:hover .wd-icon-accent {
  fill: #333333;
}

.emotion-1 .wd-icon-background {
  fill: transparent;
}

.emotion-1:hover .wd-icon-background {
  fill: transparent;
}

.emotion-1 svg {
  width: 24px;
  height: 24px;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-4 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<button
  aria-label="View All: undefined"
  className="emotion-6 emotion-7"
  role="button"
  tabIndex={0}
>
  <span
    className="emotion-0 emotion-1"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-exclamation-triangle wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M12.004 6.393L17.995 19H6z\\" class=\\"wd-icon-background\\"/><path d=\\"M12.442 3.316l8.468 16.918a.5.5 0 0 1-.448.724l-16.888-.003a.5.5 0 0 1-.447-.723l8.42-16.915a.5.5 0 0 1 .895-.001zm-.438 3.077L6 19h11.995L12.004 6.393z\\" class=\\"wd-icon-fill\\"/><path d=\\"M12.495 10c.279 0 .505.215.505.498v4.004a.494.494 0 0 1-.505.498h-.99a.496.496 0 0 1-.505-.498v-4.004c0-.275.214-.498.505-.498h.99zm0 8h-.99a.497.497 0 0 1-.505-.505v-.99c0-.291.226-.505.505-.505h.99c.291 0 .505.226.505.505v.99c0 .291-.226.505-.505.505z\\" class=\\"wd-icon-accent\\"/></g></svg>",
      }
    }
  />
  <div
    className="emotion-2 emotion-3"
  />
  <span
    className="emotion-4 emotion-5"
  >
    View All
  </span>
</button>
`;

exports[`Banner Snapshots renders a sticky banner as expected 1`] = `
.emotion-1 {
  display: inline-block;
  margin-right: 12px;
}

.emotion-1 svg {
  display: block;
}

.emotion-1 .wd-icon-fill {
  fill: #333333;
}

.emotion-1:hover .wd-icon-fill {
  fill: #333333;
}

.emotion-1 .wd-icon-accent {
  fill: #333333;
}

.emotion-1:hover .wd-icon-accent {
  fill: #333333;
}

.emotion-1 .wd-icon-background {
  fill: transparent;
}

.emotion-1:hover .wd-icon-background {
  fill: transparent;
}

.emotion-1 svg {
  width: 24px;
  height: 24px;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-6 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 500;
  color: #333333;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  background-color: #ffa126;
  box-sizing: border-box;
  padding: 8px 16px;
  text-align: left;
  border: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: background-color 120ms;
  transition: background-color 120ms;
  background-color: #ffa126;
  color: #333333;
  border-radius: 4px 0 0 4px;
  width: 222px;
}

.emotion-6:focus {
  outline: none;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-6:hover {
  cursor: pointer;
}

.emotion-6:hover {
  background-color: #f38b00;
}

.emotion-4 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  display: none;
}

<button
  aria-label="View All: undefined"
  className="emotion-6 emotion-7"
  role="button"
  tabIndex={0}
>
  <span
    className="emotion-0 emotion-1"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-exclamation-triangle wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M12.004 6.393L17.995 19H6z\\" class=\\"wd-icon-background\\"/><path d=\\"M12.442 3.316l8.468 16.918a.5.5 0 0 1-.448.724l-16.888-.003a.5.5 0 0 1-.447-.723l8.42-16.915a.5.5 0 0 1 .895-.001zm-.438 3.077L6 19h11.995L12.004 6.393z\\" class=\\"wd-icon-fill\\"/><path d=\\"M12.495 10c.279 0 .505.215.505.498v4.004a.494.494 0 0 1-.505.498h-.99a.496.496 0 0 1-.505-.498v-4.004c0-.275.214-.498.505-.498h.99zm0 8h-.99a.497.497 0 0 1-.505-.505v-.99c0-.291.226-.505.505-.505h.99c.291 0 .505.226.505.505v.99c0 .291-.226.505-.505.505z\\" class=\\"wd-icon-accent\\"/></g></svg>",
      }
    }
  />
  <div
    className="emotion-2 emotion-3"
  />
  <span
    className="emotion-4 emotion-5"
  >
    View All
  </span>
</button>
`;

exports[`Banner Snapshots renders an alert banner as expected 1`] = `
.emotion-6 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 500;
  color: #333333;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  background-color: #ffa126;
  box-sizing: border-box;
  padding: 8px 16px;
  text-align: left;
  border: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: background-color 120ms;
  transition: background-color 120ms;
  background-color: #ffa126;
  color: #333333;
  border-radius: 4px;
  width: 328px;
}

.emotion-6:focus {
  outline: none;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-6:hover {
  cursor: pointer;
}

.emotion-6:hover {
  background-color: #f38b00;
}

.emotion-1 {
  display: inline-block;
  margin-right: 12px;
}

.emotion-1 svg {
  display: block;
}

.emotion-1 .wd-icon-fill {
  fill: #333333;
}

.emotion-1:hover .wd-icon-fill {
  fill: #333333;
}

.emotion-1 .wd-icon-accent {
  fill: #333333;
}

.emotion-1:hover .wd-icon-accent {
  fill: #333333;
}

.emotion-1 .wd-icon-background {
  fill: transparent;
}

.emotion-1:hover .wd-icon-background {
  fill: transparent;
}

.emotion-1 svg {
  width: 24px;
  height: 24px;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-4 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<button
  aria-label="View All: undefined"
  className="emotion-6 emotion-7"
  role="button"
  tabIndex={0}
>
  <span
    className="emotion-0 emotion-1"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-exclamation-triangle wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M12.004 6.393L17.995 19H6z\\" class=\\"wd-icon-background\\"/><path d=\\"M12.442 3.316l8.468 16.918a.5.5 0 0 1-.448.724l-16.888-.003a.5.5 0 0 1-.447-.723l8.42-16.915a.5.5 0 0 1 .895-.001zm-.438 3.077L6 19h11.995L12.004 6.393z\\" class=\\"wd-icon-fill\\"/><path d=\\"M12.495 10c.279 0 .505.215.505.498v4.004a.494.494 0 0 1-.505.498h-.99a.496.496 0 0 1-.505-.498v-4.004c0-.275.214-.498.505-.498h.99zm0 8h-.99a.497.497 0 0 1-.505-.505v-.99c0-.291.226-.505.505-.505h.99c.291 0 .505.226.505.505v.99c0 .291-.226.505-.505.505z\\" class=\\"wd-icon-accent\\"/></g></svg>",
      }
    }
  />
  <div
    className="emotion-2 emotion-3"
  />
  <span
    className="emotion-4 emotion-5"
  >
    View All
  </span>
</button>
`;

exports[`Banner Snapshots renders an error banner as expected 1`] = `
.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-4 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-6 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 500;
  color: #333333;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  background-color: #ffa126;
  box-sizing: border-box;
  padding: 8px 16px;
  text-align: left;
  border: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: background-color 120ms;
  transition: background-color 120ms;
  background-color: #de2e21;
  color: #ffffff;
  border-radius: 4px;
  width: 328px;
}

.emotion-6:focus {
  outline: none;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-6:hover {
  cursor: pointer;
}

.emotion-6:hover {
  background-color: #a31b12;
}

.emotion-1 {
  display: inline-block;
  margin-right: 12px;
}

.emotion-1 svg {
  display: block;
}

.emotion-1 .wd-icon-fill {
  fill: #ffffff;
}

.emotion-1:hover .wd-icon-fill {
  fill: #ffffff;
}

.emotion-1 .wd-icon-accent {
  fill: #ffffff;
}

.emotion-1:hover .wd-icon-accent {
  fill: #ffffff;
}

.emotion-1 .wd-icon-background {
  fill: transparent;
}

.emotion-1:hover .wd-icon-background {
  fill: transparent;
}

.emotion-1 svg {
  width: 24px;
  height: 24px;
}

<button
  aria-label="View All: undefined"
  className="emotion-6 emotion-7"
  role="button"
  tabIndex={0}
>
  <span
    className="emotion-0 emotion-1"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-exclamation-circle wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"9\\" class=\\"wd-icon-background\\"/><path d=\\"M2 12C2 6.477 6.476 2 12 2c5.523 0 10 4.476 10 10 0 5.523-4.476 10-10 10-5.523 0-10-4.476-10-10zm2 0a8 8 0 0 0 8 8 8 8 0 0 0 8-8 8 8 0 0 0-8-8 8 8 0 0 0-8 8z\\" class=\\"wd-icon-fill\\"/><path d=\\"M12.495 7c.279 0 .505.233.505.503v5.994a.497.497 0 0 1-.505.503h-.99a.509.509 0 0 1-.505-.503V7.503c0-.278.214-.503.505-.503h.99zm0 10h-.99a.497.497 0 0 1-.505-.505v-.99c0-.291.226-.505.505-.505h.99c.291 0 .505.226.505.505v.99c0 .291-.226.505-.505.505z\\" class=\\"wd-icon-accent\\"/></g></svg>",
      }
    }
  />
  <div
    className="emotion-2 emotion-3"
  />
  <span
    className="emotion-4 emotion-5"
  >
    View All
  </span>
</button>
`;

exports[`Banner Snapshots renders as expected 1`] = `
.emotion-6 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 500;
  color: #333333;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  background-color: #ffa126;
  box-sizing: border-box;
  padding: 8px 16px;
  text-align: left;
  border: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: background-color 120ms;
  transition: background-color 120ms;
  background-color: #ffa126;
  color: #333333;
  border-radius: 4px;
  width: 328px;
}

.emotion-6:focus {
  outline: none;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-6:hover {
  cursor: pointer;
}

.emotion-6:hover {
  background-color: #f38b00;
}

.emotion-1 {
  display: inline-block;
  margin-right: 12px;
}

.emotion-1 svg {
  display: block;
}

.emotion-1 .wd-icon-fill {
  fill: #333333;
}

.emotion-1:hover .wd-icon-fill {
  fill: #333333;
}

.emotion-1 .wd-icon-accent {
  fill: #333333;
}

.emotion-1:hover .wd-icon-accent {
  fill: #333333;
}

.emotion-1 .wd-icon-background {
  fill: transparent;
}

.emotion-1:hover .wd-icon-background {
  fill: transparent;
}

.emotion-1 svg {
  width: 24px;
  height: 24px;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-4 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<button
  aria-label="View All: undefined"
  className="emotion-6 emotion-7"
  role="button"
  tabIndex={0}
>
  <span
    className="emotion-0 emotion-1"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-exclamation-triangle wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M12.004 6.393L17.995 19H6z\\" class=\\"wd-icon-background\\"/><path d=\\"M12.442 3.316l8.468 16.918a.5.5 0 0 1-.448.724l-16.888-.003a.5.5 0 0 1-.447-.723l8.42-16.915a.5.5 0 0 1 .895-.001zm-.438 3.077L6 19h11.995L12.004 6.393z\\" class=\\"wd-icon-fill\\"/><path d=\\"M12.495 10c.279 0 .505.215.505.498v4.004a.494.494 0 0 1-.505.498h-.99a.496.496 0 0 1-.505-.498v-4.004c0-.275.214-.498.505-.498h.99zm0 8h-.99a.497.497 0 0 1-.505-.505v-.99c0-.291.226-.505.505-.505h.99c.291 0 .505.226.505.505v.99c0 .291-.226.505-.505.505z\\" class=\\"wd-icon-accent\\"/></g></svg>",
      }
    }
  />
  <div
    className="emotion-2 emotion-3"
  />
  <span
    className="emotion-4 emotion-5"
  >
    View All
  </span>
</button>
`;
